src/data-model/shared/journalentrylink.ts

```diff
@@ -3,4 +3,5 @@ export class JournalEntryLink {
   journalId: string = "";
   /** compendium pack (if it's in a compendium) */
   journalPack: string = "";
+  journalName: string = "";
```

Comment:
It would be better to not use JournalEntryLink. That was a mistake from my side since I didn't know about uuid at that time. If you have a uuid there is also a function in `util.uuid2Name` that gives the name and some extra text (that might need a little refactoring). That way there is no need to store the name of a linked journalEntry.

Reply:
> use uuid instead of the JournalEntryLink interface.

So we would have to create our own handler to look for the data-uuid attribute?  Or does the built-in journal link functionality handle that somehow?  

> homeland + city

If we make the homeland just be "Sartar" then we could do a list of links to regions the way it does tribes and clans right now, but then where do we put the tribes and clans and the rest of the stuff?  All of the fields in the Homeland tab actually pertain to the "Region or City" and (with the exception of Cults which is some from Homeland and some from Region).  Maybe the Homeland tab should really have been called "Region" the way I have it now.

We could have a "Region" item or structure that is a child of homeland and has a link to the journal and all of the fields on Homeland the Homeland tab and they get repeated for each "Region".  Sartar has 12 Regions, so I think the form for that would be unwieldy.  Also, if the Actor sheet is going to hang on to the Homeland object after character creation so that the player has a handy reference to things they might know, I think it makes sense for someone from Colymar to know a lot of detailed information about those lands but if they have links to things in Alone and Wilmskirk and so on, it's both too much information for the player and probably might be more than their character would reasonably know.

In the Wizard I was envisioning that we'd look through the journals (in both the world and all compendia?) for Homeland objects and group them by the data.homeland values, so we could present a list like "Grazelands, Lunar Tarsh, Prax, Sartar, etc".  The player picks one of those, and then we look for journals with that data.homeland and present a list based on the data.region, like "Alda-Chur, Alone, Boldhome, Colymar Tribe, etc".  

I agree that it creates a lot of Homeland objects, 12 for Sartar, but I think it keeps them in more manageable chunks of information, especially since the bulk of the really important stuff derives from the Region.  Everything about the overall Homeland (like Sartar) is pretty much free-form text that can be put into a shared journal article.

Reply:
> So we would have to create our own handler to look for the data-uuid attribute? Or does the built-in journal link functionality handle that somehow?

When an item is dropped, the uuid to to that item is included in the provided data. If you have the uuid you can lookup the name (see util.uuid2Name() ) so there is no need to store that.  And for generating the html a uuidLink.hbs would do it, there is some functionality built in to Foundry in [enrichHTML](https://foundryvtt.com/api/TextEditor.html#.enrichHTML) but that is more for parsing journal style links like `@Compendium[rqg.demo-actors.ro196XqFnrNvYB69]{Demo Spirit}` to html links with icon.

The weapon item already use uuid. That said I agree that this can be looked at later (even though it hurts a little to add more fields to all items that use the journalLink interface already. The important thing is to get to the point where this can be tested in a simple character creator.

> I agree that it creates a lot of Homeland objects, 12 for Sartar, but I think it keeps them in more manageable chunks of information, especially since the bulk of the really important stuff derives from the Region. Everything 

Yes after your explanation I agree with you. The character creation wizard would include a specialised version of a compendium browser (using getItemByRqid). First display a list of Homelands (derived from the list of all homeland items) and then after one of them is selected, show either the actual homeland items or a dropdown with regions.
I think we need one more "find rqid" function that returns a list so that one can get the homelands to choose from (but that's in another case...).